_episodes/06-plotting.md

```diff
@@ -0,0 +1,341 @@
+- "How can I save my plot for publishing?"
+objectives:
+- "Import a library into a program using `import libraryname`."
+- "Create a basic scatter plot showing relationship between two series."
+- "Create a histogram showing distribution of a series."
+- "Add a title and labels for the x-axis and y-axis."
+keypoints:
+- "[`matplotlib`](https://matplotlib.org/) is the most widely used scientific plotting library in Python."
+- "We can plot data directly from a `pandas` dataframe ([official documentation](https://pandas.pydata.org/docs/user_guide/visualization.html) )."
+- "Many kinds and styles of plots are available."
+- "We can save a plot to a file using the `savefig` function."
+- "We can create publication-quality plots with `xlabel`, `ylabel`, `title`, `legend`."
+---
+
+## Using a Library for Scientific Plotting
+
+A library is a collection of code units (functions, classes, variables) which
+implement a set of related tasks.
+In Python, [`matplotlib`](https://matplotlib.org/) is the most widely used
+library for scientific plotting.
+We commonly use its `pyplot` module and import it as follows:
+
+~~~
+import matplotlib.pyplot as plt
+~~~
+{: .language-python}
+
+In Python, a module is an object which contains Python code and which can be
+imported inside another Python program.
+Simple plots are then (fairly) simple to create:
+
+~~~
+time = [0, 1, 2, 3]
+position = [0, 100, 200, 300]
+
+plt.plot(time, position)
+plt.xlabel('Time (hr)')
+plt.ylabel('Position (km)')
+~~~
+{: .language-python}
+
+![Simple Position-Time Plot](../fig/06_simple_position_time_plot.png)
+
+The Jupyter notebook renders plots inline by default.
+
+> ## Displaying All Open Figures
+> 
+> In our Jupyter notebook example, running the cell should generate the figure
+> directly below the code. 
+> However, other Python environments (such as an interactive Python session
+> started from a terminal or a Python script executed via the command line)
+> require an additional command to display the figure:
+>
+> ~~~
+> plt.show()
+> ~~~
+> {: .language-python}
+>
+> This command can also be used within a notebook -- for instance, to display
+> multiple figures if several are created by a single cell.
+>
+{: .callout}
+
+## Plotting Data Directly from a Dataframe
+
+In Python, [`pandas`](https://pandas.pydata.org/) is is the most widely used
+library for data analysis and manipulation. Notably, it provides a very useful
+data structure called a *dataframe*. Dataframes contain tabular data which
+are:
+
+* two-dimensional;
+* potentially heterogeneous;
+* with labelled axes (rows and columns).
+
+To showcase the power and ease of use of `pandas`, let's try it on some
+oceanographic open data:
+https://data.gov.ie/dataset/celtic-explorer-underway
+Navigate to "available as CSV" and right-click the "Download" button (to copy
+the URL).
+
+~~~
+url = "http://atlas.marine.ie/midata/OceanFeatures/Celtic_Explorer_Underway.csv.zip"
+
+import pandas as pd
+
+df = pd.read_csv(url)
```

Comment:
maybe you can also mention that it is possible to load the data from the computer using pandas

Reply:
maybe you should shortly add a sentence on what is a .csv format. And if they have a problem to load one of theirs properly to check 'delimiter' option.
and.. perhaps show them how to use help to understand what function is doing

Reply:
Ok. Yes! I want to show how to get help when I introduce built-in functions, but I'll show it here again.

Reply:
Done b449c5762f. I'll mention the 'delimiter' parameter only when speaking.